blockend/picocard.js

```diff
@@ -43,15 +40,30 @@ const { compressToUint8Array, decompressFromUint8Array } = lzString
 // 10kB accurate version: https://github.com/mathiasbynens/emoji-regex/blob/master/index.js
 export const EMOJI_REGEXP = /!([FLDd~|/.-]{0,4})([^!\n .}]{1,8})!/
 
+/*
+ * encode/decode functions works like old pack/unpack before we introduced
+ * picostack.
+ * They are used only for drafts ATM.
+ * They handle compression & encryption types and attempt to be space-efficient
+ */
+export function encode (card, secret) {
+  if (Buffer.isBuffer(card)) throw new Error('Expected a card object')
+  return msgPack(pack(card, secret))
+}
+export function decode (buffer, secret) {
+  if (!Buffer.isBuffer(buffer)) throw new Error('Expected binary Buffer')
+  const props = msgUnpack(buffer)
+  return unpack(props, secret)
+}
+
+/**
+ * Minifies object-keys and compresses/encrypts the card-contents
+ * If used outside of pico-stack use msgpack (unpack) to read
+ * it.
+ */
 export function pack (props, secret) {
```

Comment:
```js
export function pack (props, secret)
```
`secret` is not used inside `pack()` should be safe to remove if not used. 

Since we are moving back `decrypt` we could move `encrypt` here as well, only problem is that we either have to store secret in `$draft` or prompt user to input pin on publish. This might be a good solution since it removes at least one major bug: 
`when encrypting rant and not publishing: encrypted text is saved in draft which leaves the user with a useless piece of encrypted text that cannot be decrypted`

Reply:
Yeah, that is a bad bug that would be nice to avoid.. I think the current implementation you did is a nice tradeoff between both problems. I need to research some more..